pkg/estimator/server/framework/runtime/framework_test.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2024 The Karmada Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package runtime
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"testing"
+
+	"github.com/karmada-io/karmada/pkg/estimator/pb"
+	"github.com/karmada-io/karmada/pkg/estimator/server/framework"
+)
+
+var (
+	errEstimateReplica = fmt.Errorf("estimate failed")
+)
+
+type estimateReplicaResult struct {
+	replica int32
+	err     error
+}
+
+type injectedResult struct {
+	estimateReplicaResult estimateReplicaResult
+}
+
+// TestPlugin implements all Plugin interfaces.
+type TestPlugin struct {
+	name string
+	inj  injectedResult
+}
+
+func (pl *TestPlugin) Name() string {
+	return pl.name
+}
+
+func (pl *TestPlugin) Estimate(ctx context.Context, replicaRequirements *pb.ReplicaRequirements) (int32, error) {
```

Comment:
there is a lint error which advices changing this line to:

```suggestion
func (pl *TestPlugin) Estimate(_ context.Context, _ *pb.ReplicaRequirements) (int32, error) {
```